### PR TITLE
No ack delivery for prio messages, #21371

### PR DESCRIPTION
**Pull request by patriknw**
- and send prio messages enclosed in actor selection
  over the control stream

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #21371

File: akka-remote/src/main/scala/akka/remote/artery/Association.scala [hunk 1]
```diff
@@ -287,12 +287,18 @@ private[remote] class Association(
...
       message match {
-        case _: SystemMessage | ClearSystemMessageDelivery | _: ControlMessage ⇒
+        case _: SystemMessage ⇒
           val outboundEnvelope = createOutboundEnvelope()
           if (!controlQueue.offer(createOutboundEnvelope())) {
             quarantine(reason = s"Due to overflow of control queue, size [$controlQueueSize]")
             dropped(ControlQueueIndex, controlQueueSize, outboundEnvelope)
           }
+        case ActorSelectionMessage(_: PriorityMessage, _, _) | _: ControlMessage | ClearSystemMessageDelivery ⇒
+          // ActorSelectionMessage with PriorityMessage is used by cluster and remote failure detector heartbeating
```
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/artery/Association.scala]: "// so those cannot be matched on actor path" ?

Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/artery/Association.scala]: NVM, I misstook where in the case statement this is..

Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/Association.scala]: yes, all ActorSelection messages go to the same intermediate destination

**Comment by johanandren** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM

**Comment by drewhk** [comment]
LGTM

**Pull request merged**